### ISSUE TITLE
Can't find variable: grecaptcha

### ISSUE THREAD

**Issue by sashsvamir** [himiklab/yii2-recaptcha-widget#58]
If you requested ajax page with captcha from page with not ready loading api.js (just page without any recaptchas), js get error: **ReferenceError: Can't find variable: grecaptcha**.
Probably this error going from here:
https://github.com/himiklab/yii2-recaptcha-widget/blob/bc9df57a2b420aa7deaec4f6556ff79d57fa435b/ReCaptcha.php#L122
But real problem i think is to api must be loading before executes this part of code.

**Comment by himiklab** [comment]
Are you using `renderAjax`?

**Comment by sashsvamir** [comment]
Yes, requested path going to action of controller, where returned view with form:
```php
return $this->renderAjax('feedback', ['model' => $model]);
```
form contained follow line:
```php
$form->field($model, 'verifyCode')->widget(ReCaptcha::className())->label(false) ?>
```

**Comment by himiklab** [comment]
And version `yii2-recaptcha-widget` is 1.1.0? It have ajax fixes.

**Comment by sashsvamir** [comment]
just now i'm execute:
```sh
composer remove sashsvamir/yii2-recaptcha-widget
composer clearcache
composer require himiklab/yii2-recaptcha-widget:"~1.1.0"
```
and tried again test this issue where i still got "Can't find variable: grecaptcha".

**Comment by himiklab** [comment]
This is strange. I need a code to reproduce the error.

**Comment by sashsvamir** [comment]
ok. i'm get basic app to reproduce the error:
```sh
composer create-project --prefer-dist --stability=dev yiisoft/yii2-app-basic basic
composer require himiklab/yii2-recaptcha-widget:"~1.1.0"
```

then, add to `config/web.php`:
```php
	    'reCaptcha' => [
		    'name' => 'reCaptcha',
		    'class' => 'himiklab\yii2\recaptcha\ReCaptcha',
		    'siteKey' => '6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI', // pass test key
		    'secret' => '6LeIxAcTAAAAAGG-vFI1TnRWxMZNFuojJ4WifJWe', // pass test key
	    ],
```

next create `models/Feedback.php`:
```php
<?php

namespace app\models;

use yii\base\Model;
use himiklab\yii2\recaptcha\ReCaptchaValidator;

/**
 * Feedback
 **/
class Feedback extends Model
{
	public $verifyCode;

    /**
     * @inheritdoc
     */
    public function rules()
    {
        return [
			['verifyCode', ReCaptchaValidator::className(), 'uncheckedMessage' => 'You need pass captcha checking'],
        ];
    }

}

```

create `controllers/AjaxController.php`:
```php
<?php

namespace app\controllers;

use Yii;
use yii\web\Controller;
use app\models\Feedback;

class AjaxController extends Controller
{

	/**
	 * Displays captcha.
	 * @return string
	 */
	public function actionIndex()
	{
		$model = new Feedback();

		if ($model->load(Yii::$app->request->post()) && $model->validate()) {
			return $this->asJson([
				'result' => 'success',
			]);
		}

		return $this->renderAjax('index', ['model' => $model]);
	}

}
```

add view `views/ajax/index.php` with:
```php
<?php

/* @var $this yii\web\View */

use yii\helpers\Html;
use yii\widgets\ActiveForm;
use himiklab\yii2\recaptcha\ReCaptcha;

?>
<div class="ajax-index">


	<? $form = ActiveForm::begin([
		'id' => 'feedback-form',
	]) ?>

		<?= $form->field($model, 'verifyCode')->widget(ReCaptcha::className())->label(false) ?>

		<?= Html::submitButton('Отправить', ['class' => 'btn btn-primary']) ?>

	<? ActiveForm::end() ?>

</div>
```



and add follow js code to bottom of `site/index.php` to render ajax on click button:
```php
//...
<?
$ajaxUrl = \yii\helpers\Url::to(['ajax/index']);
// language=JS
$this->registerJs("

	$('.jumbotron a').on('click', function (event) {
		event.preventDefault();
		var parent = $(this);
		
		$.ajax('{$ajaxUrl}', {
			success: function(response) {
			    parent.after(response);
			}
		});
	});

", $this::POS_READY);
?>
```


And after click on big green "Get started with Yii" button, i got:
"Uncaught ReferenceError: grecaptcha is not defined".
But here ajax page with captcha loading, but js validation not working.

On my project where i found this error first time a'm not see ajax form because js script stopping rather than form was showing. But error still here.

**Comment by himiklab** [comment]
https://github.com/himiklab/yii2-recaptcha-widget/commit/1df2de2d6159c479f8b0260de7d34866a3571717

**Comment by sashsvamir** [comment]
Ok, this issue was resolved, but this commit lead to another unwanted behavior — now not working multiply ajax loading captcha on one page.

**Comment by himiklab** [comment]
https://github.com/himiklab/yii2-recaptcha-widget/commit/b95b3b3035558da3f2094a849d20560bfb329ee3

**Comment by sashsvamir** [comment]
Now js validation not working for first loaded (by ajax) captcha.

**Comment by himiklab** [comment]
https://github.com/himiklab/yii2-recaptcha-widget/commit/6a43b9dae398d440eac4e72153a9d982259ec45e

**Comment by sashsvamir** [comment]
now, on first ajax-loading recaptcha i got js error: `Uncaught ReferenceError: grecaptcha is not defined`. i'm create fix #59
now extension works good (in my tested enveronments).

**Comment by sashsvamir** [comment]
I suggest one more solution:
https://github.com/himiklab/yii2-recaptcha-widget/blob/6a43b9dae398d440eac4e72153a9d982259ec45e/ReCaptcha.php#L145 there callback function executes after [request succeeds](https://api.jquery.com/jquery.getscript/).
If we call this function earlier than js has parsed and loading this script we have no object `grecaptcha`.

Now we have query api.js with callback param `onload` here:
https://github.com/himiklab/yii2-recaptcha-widget/blob/6a43b9dae398d440eac4e72153a9d982259ec45e/ReCaptcha.php#L109

this query will do execute our `callback` every first time as js loading this script, next we should executes `callback` manually.

so we can remove
https://github.com/himiklab/yii2-recaptcha-widget/blob/6a43b9dae398d440eac4e72153a9d982259ec45e/ReCaptcha.php#L145

and call `recaptchaOnloadCallback()` when we have ajax request and `grecaptcha` object exist. I have updated pull request #59.
now: we have called `callback` on `api.js` loading at first request and next times when ajax request executes.
